lib/ecto/changeset/relation.ex

```diff
@@ -83,7 +83,8 @@ defmodule Ecto.Changeset.Relation do
   def cast(%{cardinality: :many} = relation, params, current, on_cast) when is_map(params) do
     params =
       params
-      |> Enum.sort_by(&elem(&1, 0))
+      |> Enum.map(&key_as_int/1)
+      |> Enum.sort
```

Comment:
Yup, `Enum.sort` is fine. Plus it will only sort by value if the keys are different.